### ISSUE TITLE
ConnectionError: ('Connection aborted.', error(104, 'Connection reset by peer'))

### ISSUE THREAD

**Issue by blinkesto** [saltstack/salt#55113]
### Description of Issue
Connection reset by peer when creating VMs with CloudClient.  I am trying to create 8 VMs and have 5 running in parallel, but one fails. 

### Setup
Calling CloudClient.map_run from python script.

### Steps to Reproduce Issue
Just running map_run with 8 VMs and parrallel set to true.  

### Versions Report
Salt Version:
           Salt: 2018.3.4
 
Dependency Versions:
           cffi: 1.12.2
       cherrypy: unknown
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: 2.0.5
      gitpython: 2.1.11
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: 2.6.1
   pycryptodome: 3.8.0
         pygit2: Not Installed
         Python: 2.7.13 (default, Mar 30 2018, 15:31:59)
   python-gnupg: 2.3.1
         PyYAML: 3.11
          PyZMQ: 18.0.1
           RAET: Not Installed
          smmap: 2.0.5
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.3.1
 
System Versions:
           dist: redhat 6.10 Santiago
         locale: UTF-8
        machine: x86_64
        release: 2.6.32-754.14.2.el6.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 6.10 Santiago

**Comment by sagetherage** [comment]
@xeacott I will need you triage this one, please. Let me know if you do not have time.

**Comment by sagetherage** [comment]
@Akm0d assigning you to triage and reproduce this issue only. Release priorities are first, but wanted your eyes on this to triage, only.

**Comment by sagetherage** [comment]
@garethgreenaway this tickets needs to be triage'd can I get your help please?

**Comment by sagetherage** [comment]
@waynew this ticket still needs to be triaged, can you look at it today, please?

**Comment by waynew** [comment]
Hi @blinkesto are you still having the problem?

Do you get any more information than this? Like a stack trace? It's not obvious from looking at the code where this would happen.

**Comment by waynew** [comment]
@blinkesto any updates?

**Comment by sagetherage** [comment]
closing this issue since we have not heard from you @blinkesto if this issue is still persistent, please comment or open another issue.

**Comment by Adamleez** [comment]
If you are accessing someone else site this indicates that the server simply dropped your connection via TCP-RST. The site you are trying to access may have issues, like its still coming up or going down, it may have other issues as well and could be a number of other things as well. So what to to;

If the site you are trying to access is not you site there is not much you can do about it. You could sent the site admin an email perhaps.

OR

If this is a site you own you should make sure everything is up and running, check log files and begin diagnostic [procedures](https://espiort.club/how-fix-cant-send-a-message-sprint-error-104-update/).
